sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/AbstractLNAdapter.java

```diff
@@ -938,20 +902,75 @@ private void removeExtRefSourceBinding(final TExtRef tExtRef) {
      * The AccessPoint must have DataSet creation capabilities
      *
      * @param dataSetName Name of the dataSet
-     * @param serviceSettingsType GOOSE, SMV or REPORT service type
+     * @param controlBlockEnum GOOSE, SMV or REPORT service type
      * @throws ScdException throws when IED does not have DataSet creation capabilities
      * @see LDeviceAdapter#hasDataSetCreationCapability
      */
-    public DataSetAdapter createDataSetIfNotExists(String dataSetName, ServiceSettingsType serviceSettingsType) {
+    public DataSetAdapter createDataSetIfNotExists(String dataSetName, ControlBlockEnum controlBlockEnum) {
         return findDataSetByName(dataSetName).orElseGet(() -> {
-            if (!getCurrentLDevice().hasDataSetCreationCapability(serviceSettingsType)) {
+            if (!getParentLDevice().hasDataSetCreationCapability(controlBlockEnum)) {
                 throw new ScdException("IED/AccessPoint does not have capability to create DataSet of type %s in %s"
-                    .formatted(serviceSettingsType, getXPath()));
+                    .formatted(controlBlockEnum, getXPath()));
             }
             TDataSet newDataSet = new TDataSet();
             newDataSet.setName(dataSetName);
             currentElem.getDataSet().add(newDataSet);
             return new DataSetAdapter(this, newDataSet);
         });
     }
+
+    /**
+     * Creates Control Block in specified LNode in current IED
+     *
+     * @param controlBlock Control Block data to add to this LN
+     * @return created <em>ControlBlockAdapter</em> object
+     * @throws ScdException throws when inconsistency between given ControlBlock and IED configuration
+     */
+    public ControlBlockAdapter addControlBlock(ControlBlock controlBlock) {
+
+        controlBlock.validateCB();
+        if (controlBlock.getSecurityEnable() != TPredefinedTypeOfSecurityEnum.NONE) {
```

Comment:
`controlBlock.getSecurityEnable() != TPredefinedTypeOfSecurityEnum.NONE` is a condition that will be checked also in `validateSecurityEnabledValue` method either the `ControlBlock` type.
```suggestion
        controlBlock.validateSecurityEnabledValue(getParentIed());
```
It is false to have `securityEnable` at `ControlBlock` class because all blocks don't materially implement it.
 I suggest keeping the `securityEnable` field only for the two `ControlBlock` types that really had it and
removing the necessity to implement Exception of type `UnsupportedOperationException` like this
```java
throw new UnsupportedOperationException("ReportControl blocks do not have 'securityEnable' attribute");
    } 
```

Reply:
OK. I removed `securityEnable` from `ControlBlock` and add it to `GooseControlBlock` and `SMVControlBlock`.